src/models/appointment.ts

```diff
@@ -0,0 +1,44 @@
+import mongoose from "mongoose";
+
+export interface IAppointment {
+    patient: string;
+    doctor: string;
+    date: Date;
+    time: string;
+    type: string;
+    link: string;
+}
+
+export interface IAppointmentModel extends IAppointment, mongoose.Document { }
+
+const appointmentSchema = new mongoose.Schema({
+    patient: {
+        type: mongoose.Schema.Types.ObjectId,
+        ref: "User",
+        required: true,
+    },
+    doctor: {
+        type: mongoose.Schema.Types.ObjectId,
+        ref: "User",
+        required: true,
+    },
+    date: {
+        type: Date,
+        required: true,
+    },
+    time: {
+        type: String,
+        required: true,
+    },
+    type: {
+        type: String,
+        required: true,
+        enum: ["online", "face2face"],
```

Comment:
Import this enum from another file such that if we change the appointment type we change it once. Also, change 'face2face' to 'onsite' (more professional)